### ISSUE TITLE
High CPU usage when idle

### ISSUE THREAD

**Issue by lukamicoder** [ultralight-ux/ultralight#276]
I've downloaded the latest SDK and compiled the sample applications with the "--config Release" option on my laptop:
OS: Linux Mint 20 Cinnamon
CPU: Intel Core i5-6440HQ
RAM: 16GB

All the sample applications run fine; however, their CPU usage continuously fluctuates between 6-8% while they are just sitting there idle. 

Is there anything I can do to reduce the CPU usage?

**Comment by adamjs** [comment]
Yes, the GLFW port of AppCore spins the run loop too greedily-- right now it's just using a very primitive approach (calling `glfwPollEvents()` and sleeping for 1 millisecond in a loop): <https://github.com/ultralight-ux/AppCore/blob/master/src/linux/AppGLFW.cpp#L145>

A better approach is implemented in Sample 7 (which uses scheduled paints and wakes up early when the OS sends an input event via `glfwWaitEventsTimeout()`): <https://github.com/ultralight-ux/Ultralight/blob/master/samples/Sample%207%20-%20OpenGL%20Integration/src/Sample.cpp#L164>

I'll apply these optimizations to AppCore.
